src/jit/codegencommon.cpp

```diff
@@ -6722,11 +6722,11 @@ void CodeGen::genZeroInitFltRegs(const regMaskTP& initFltRegs, const regMaskTP&
                     inst_RV_RV(INS_vmov_i2f, reg, initReg, TYP_FLOAT, EA_4BYTE);
                 }
 #elif defined(_TARGET_XARCH_)
-                // Xorpd xmmreg, xmmreg is the fastest way to initialize a float register to
+                // Xorps xmmreg, xmmreg is the fastest way to initialize a float register to
                 // zero instead of moving constant 0.0f.  Though we just need to initialize just the 32-bits
                 // we will use xorpd to initialize 64-bits of the xmm register so that it can be
                 // used to zero initialize xmm registers that hold double values.
-                inst_RV_RV(INS_xorpd, reg, reg, TYP_DOUBLE);
+                inst_RV_RV(INS_xorps, reg, reg, TYP_DOUBLE);
```

Comment:
I admit I don't fully understand the codegen part, but is Typ_Double still correct for the float instruction?

Reply:
Hmm, I'll check. Strictly speaking both `TYP_DOUBLE` and `TYP_FLOAT` are incorrect here as the instruction operates on a 16 byte XMM register. But it shouldn't matter to the emitter.

Reply:
The only type that makes a difference here is `TYP_SIMD32`, that would give you `vxorps ymm0, ymm0` instead of `(v)xorps xmm0, xmm0`.

I'll leave it as is because there are other similar mismatches in the code. For example, above there's
`inst_RV_RV(ins_Copy(TYP_DOUBLE), reg, dblInitReg, TYP_DOUBLE);`. `ins_Copy(TYP_DOUBLE)` is really `INS_movaps` and again you have a "ps" instruction with `TYP_DOUBLE`.

Reply:
Thank you for your explanation